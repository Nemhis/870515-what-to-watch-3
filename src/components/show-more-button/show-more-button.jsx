import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {ActionCreator, SHOWN_MOVIES_COUNT} from '../../actions';

const ShowMoreButton = ({isVisible, onClick, shownMoviesCount}) => {
  return isVisible ?
    <div className="catalog__more">
      <button className="catalog__button" type="button" onClick={() => {
        onClick(shownMoviesCount + SHOWN_MOVIES_COUNT);
      }}>Show more</button>
    </div>
    :
    null;
};

ShowMoreButton.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  shownMoviesCount: PropTypes.number.isRequired,
};

const mapStateToProps = ({filmsCount, shownFilms, shownMoviesCount}) => ({
  isVisible: filmsCount > shownFilms.length,
  shownMoviesCount,
});

const mapDispatchToProps = (dispatch) => ({
  onClick(shownMoviesCount) {
    dispatch(ActionCreator.changeShownFilmsCount(shownMoviesCount));
    dispatch(ActionCreator.sliceFilmsByShownCount());
  }
});

export {ShowMoreButton};
export default connect(mapStateToProps, mapDispatchToProps)(ShowMoreButton);
