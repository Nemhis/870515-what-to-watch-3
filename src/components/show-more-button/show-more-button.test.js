import React from 'react';
import renderer from 'react-test-renderer';

import {ShowMoreButton} from './show-more-button.jsx';


describe(`Render correctly <ShowMoreButton>`, () => {
  it(`Should render when visible is true`, () => {
    const tree = renderer
      .create(<ShowMoreButton isVisible={true} onClick={() => {}} shownMoviesCount={1} />).toJSON(`test`);

    expect(tree).toMatchSnapshot();
  });

  it(`Shouldn't render when visible is false`, () => {
    const tree = renderer
      .create(<ShowMoreButton isVisible={false} onClick={() => {}} shownMoviesCount={1} />).toJSON(`test`);

    expect(tree).toMatchSnapshot();
  });
});
