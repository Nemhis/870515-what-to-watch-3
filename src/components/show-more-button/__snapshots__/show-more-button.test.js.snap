// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly <ShowMoreButton> Should render when visible is true 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;

exports[`Render correctly <ShowMoreButton> Shouldn't render when visible is false 1`] = `null`;
